app/assets/javascripts/application.js

```diff
@@ -47,7 +50,6 @@
 //= require bootstrap-select
 //= require bootstrap-hover-dropdown
 //= require c3/c3
-//= require d3/d3
```

Comment:
> Sorry, could we keep c3 and d3 together? And you still have single quotes in the Gemfile, albeit only for version now.

will do. will take c3 up to d3. apparently the order in this file matters. having d3 down next to c3 causes the topology not to be drawn..